builtin/providers/aws/provider.go

```diff
@@ -113,6 +113,7 @@ func Provider() terraform.ResourceProvider {
 		DataSourcesMap: map[string]*schema.Resource{
 			"aws_ami":                dataSourceAwsAmi(),
 			"aws_availability_zones": dataSourceAwsAvailabilityZones(),
+			"aws_s3_object":          dataSourceAwsS3Object(),
```

Comment:
Should we instead call this `aws_s3_bucket_object` so it matches with the resource of the same name? So far we've been trying to make the data sources match the corresponding resources as much as possible (in terms of attributes too) so it's relatively easy to switch between them.